plugins/ltac/tacinterp.ml

```diff
@@ -1125,17 +1132,18 @@ and eval_tactic ist tac : unit Proofview.tactic = match tac with
   | TacSelect (sel, tac) -> Tacticals.New.tclSELECT sel (interp_tactic ist tac)
   (* For extensions *)
   | TacAlias (loc,(s,l)) ->
-      let (ids, body) = Tacenv.interp_alias s in
+      let alias = Tacenv.interp_alias s in
+      Option.iter (fun o -> warn_deprecated_alias ?loc (s,o)) @@ alias.Tacenv.alias_deprecation;
```

Comment:
This is not the right place to check for alias deprecation. This will run a warning at runtime, which is not what you want. Move this code to `Tacintern.intern_tactic_seq` in the `TacAlias` case. This will hopefully also solve the problems you have with the warning not firing.

Reply:
But is it ok to call `Tacenv.interp_alias` in `Tacintern.intern_tactic_seq`?

Reply:
It is.